crystal-map-connector/src/main/kotlin/ch/ergon/dope/extension/type/ObjectField.kt

```diff
@@ -0,0 +1,52 @@
+package ch.ergon.dope.extension.type
+
+import ch.ergon.dope.resolvable.expression.unaliased.type.Field
+import ch.ergon.dope.toDopeType
+import ch.ergon.dope.validtype.ObjectType
+import com.schwarz.crystalapi.schema.CMJsonField
+import com.schwarz.crystalapi.schema.CMJsonList
+import com.schwarz.crystalapi.schema.CMObjectField
+import com.schwarz.crystalapi.schema.CMObjectList
+import com.schwarz.crystalapi.schema.CMType
+import com.schwarz.crystalapi.schema.Schema
+import kotlin.reflect.KProperty1
+
+class ObjectField<S : Schema>(val schema: S, val name: String, val path: String) : Field<ObjectType>(name, path)
+
+inline fun <reified T : CMType, S : Schema> ObjectField<S>.getField(field: KProperty1<S, T>): T {
+    val schemaField = field.get(schema)
+    val newPath = if (path.isBlank()) name else "$path`.`$name"
+    return when (schemaField) {
+        is CMJsonField<*> -> CMJsonField<Any>(schemaField.name, newPath) as T
+        is CMJsonList<*> -> CMJsonList<Any>(schemaField.name, newPath) as T
+        is CMObjectField<*> -> CMObjectField(schemaField.element, schemaField.name, newPath) as T
+        is CMObjectList<*> -> CMObjectList(schemaField.element, schemaField.name, newPath) as T
+        else -> error("Unsupported CMType: $schemaField")
+    }
+}
+
+/**
+ * Retrieves a schema attribute of a specific type from the object field's schema.
```

Comment:
Shouldn't this documentation be above the funtion above? Would documentate both

Reply:
added it to the other one as well